src/main/java/com/quorum/gauge/ext/EnhancedClientTransactionManager.java

```diff
@@ -44,16 +50,31 @@
 
     private Quorum quorum;
 
+    private Web3jService web3j;
+
     public EnhancedClientTransactionManager(Quorum quorum, String fromAddress, String privateFrom, List<String> privateFor, List<PrivacyFlag> contractFlag, int attempts, int sleepDuration) {
         super(quorum, fromAddress, privateFrom, privateFor, attempts, sleepDuration);
         this.quorum = quorum;
         this.contractFlag = contractFlag;
+
+        try {
+            JsonRpc2_0Web3j q = (JsonRpc2_0Web3j) quorum;
+            Field f = JsonRpc2_0Web3j.class.getDeclaredField("web3jService"); //NoSuchFieldException
+            f.setAccessible(true);
```

Comment:
is there no API to get this client except reflection? I think there is a service in the QAT framework that has the Web3jService

Reply:
I don't believe so - the services create EnhancedClientTransactionManagers as and when they need them

Reply:
is there no way to inject this? or obtain from the Node? I'm reluctant to use reflection, as this it's a matter of changing our architecture

Reply:
you have `QuorumNodeConnectionFactory.getWeb3jService(..)` and `QuorumNodeConnectionFactory.getWeb3jConnection(..)`

Reply:
connectionFactory isn't available to the TransactionManagers as they aren't spring services - the constructor could be altered to also inject web3jservice but that creates a lot of noise as the constructors are used extensively throughout the test code

Reply:
ok, lets keep it as it is then. got merged anyway